### ISSUE TITLE
[COMPATIBILITY] [1.20.4] Incompatible with Iris 1.7

### ISSUE THREAD

**Issue by tytytyt23** [Exopandora/ShoulderSurfing#166]
### Description

whenever i try to run mc with this mod it crashes , and yes i installed the forgeconfig mod . i ran it in iris + fabric + distant horizons

### Steps to Reproduce

1) open mc launcher
2)click play

### Expected Behavior

i expected everything to go as planned (expected it to work)

### Screenshots

![Screenshot 2024-03-31 094517](https://github.com/Exopandora/ShoulderSurfing/assets/165542586/8fab2855-40e8-4f90-a045-e2205ba2e12d)

### Game Logs

i cant find it idk why

### Additional Information

mods i got installed that worked before installing forgeconfig and Shoulder Surfing mod

1) autohud
2) chunky
3) indium
4) sodium
5) modmenu
6) distant horizons
7) iris
8) fabric

**Comment by Exopandora** [comment]
You can find the crash logs in `%appdata%\.minecraft\crash-logs`. Please upload the file with the latest date as its name. If the folder  ordoes file not exist, please use the latest log instead. It can be found in `%appdata%\.minecraft\logs` and is called `latest.log`. Please also provide your mnecraft version and the versions of all mods used.

**Comment by YuJianghan** [comment]
I have also encountered this issue. The latest test versions of Iris Shaders and Distant Horizons that support shaders seem to be the problem. After troubleshooting, it appears that using the test version of Iris Shaders together with Shoulder Surfing Reloaded causes a crash at game startup.

Game version:
Minecraft_1.20.4_Fabric_0.15.7

Mod list:
- fabric-api-0.96.11+1.20.4
- sodium-fabric-0.5.8+mc1.20.4
- iris-1.7.0-snapshot+mc1.20.4-482321c
- ShoulderSurfing-Fabric-1.20.4-2.9.6
- ForgeConfigAPIPort-v20.4.3-1.20.4-Fabric

I also tried the latest official version of Iris Shaders, iris-mc1.20.4-1.6.17.jar, and the game did not crash. Furthermore, Distant Horizons does not seem to be related to the crash. The cause of the crash is specifically the test version of Iris Shaders. The download link for the test version of Iris Shaders is here: https://github.com/IrisShaders/Iris-Installer-Files/releases/download/1.6.11/Iris-Sodium-Beta-1.20.4.zip

Here is my log file:
[latest.log](https://github.com/Exopandora/ShoulderSurfing/files/14816393/latest.log)

**Comment by ExodusOTH** [comment]
Just wanted to say as a developer I am impressed by the quick turnaround on this bug, and grateful for the timing. Hopeful to see the next version on CurseForge soon! 

Also, are there plans for Shoulder Surfing 3.0 to support Minecraft 1.20.1? I built locally so I could use it in the short term and was met with a dependency error saying it required 1.20.2 or higher. Unfortunately 1.20.1 is the most widely supported version for my modpacks currently, so I figured I'd ask.
